src/plugins/telemetry/server/get_next_attempt_date.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { randomInt } from 'crypto';
+import moment from 'moment';
+import { REPORT_INTERVAL_BUFFER_MS } from '../common/constants';
+
+const REPORT_INTERVAL_BUFFER_S = REPORT_INTERVAL_BUFFER_MS / 1000;
+
+/**
+ * Returns the next attempt to send telemetry
+ * @param fromMs The last attempt in ms from epoch
+ */
+export function getNextAttemptDate(fromMs: number): Date {
+  const dayPlus24hours = moment(fromMs).utcOffset(0).add(24, 'hours');
+  const endOfNextDay = dayPlus24hours.clone().endOf('day');
+  let nextAttemptDate = dayPlus24hours
+    .clone()
+    .add(randomInt(-REPORT_INTERVAL_BUFFER_S, REPORT_INTERVAL_BUFFER_S), 'seconds');
```

Comment:
Using seconds instead of milliseconds because we want clear differentiation between 2 instances attempting to report telemetry. If the `randomInt` returned milliseconds, there are higher chances that multiple instances would be only hundreds of milliseconds away.